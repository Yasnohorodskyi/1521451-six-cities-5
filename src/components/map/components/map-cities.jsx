import React, {useState, useEffect} from 'react';
import {Map, TileLayer} from "react-leaflet";
import L from 'leaflet';
import PropTypes from 'prop-types';

import {offerItem} from '../../../shapes/offer-item';
import {currentCityShape} from '../../../shapes/current-city';
import {ZOOM} from '../const';

const MapCities = ({currentCity, offers}) => {
  let [map, setMap] = useState(null);

  const mapRef = React.createRef();

  const mapSettings = (info) => {

    const icon = L.icon({
      iconUrl: `/img/pin.svg`,
      iconSize: [30, 30]
    });

    const zoom = ZOOM;

    let newMap = L.map(`map`, {
      center: [
        info[0].city[`location`].latitude,
        info[0].city[`location`].longitude
      ],
      zoom,
      zoomControl: false,
      marker: true
    });

    newMap.setView([info[0].city[`location`].latitude, info[0].city[`location`].longitude], zoom);

    L.tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
      attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
    })
      .addTo(newMap);


    info.forEach((offer) => {
      const offerCords = [offer.location.latitude, offer.location.longitude];
      L.marker(offerCords, {icon}).addTo(newMap)._icon.id = `marker-${offer.id}`;
    });

    return newMap;
  };

  useEffect(() => {

    if (!map) {
      setMap(
          mapSettings(offers)
      );
    } else {
      map.remove();

      setMap(
          mapSettings(offers)
      );

    }

  }, [currentCity]);

  return (
    <div id='map' ref={mapRef}>
      <Map
        doubleClickZoom={true}
        scrollWheelZoom={true}
        dragging={true}
        animate={true}
        easeLinearity={0.35}
      >
        <TileLayer
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          url='http://{s}.tile.osm.org/{z}/{x}/{y}.png'
        />
      </Map>
    </div>
  );
};

MapCities.propTypes = {
  currentOffer: PropTypes.number,
  currentCity: currentCityShape,
  offers: PropTypes.arrayOf(
      offerItem
  ),
  cities: PropTypes.arrayOf(
      currentCityShape
  ),
  max: PropTypes.number
};

export {MapCities};
