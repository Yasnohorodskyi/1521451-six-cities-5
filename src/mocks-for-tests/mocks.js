export const offers = [
  {
    city: {
      name: `Hamburg`,
      location: {
        latitude: 53.550343,
        longitude: 10.000654,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`
    ],
    title: `The house among olive `,
    isFavorite: true,
    isPremium: false,
    rating: 4.3,
    type: `house`,
    bedrooms: 5,
    maxAdults: 6,
    price: 190,
    goods: [
      `Washer`,
      `Breakfast`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 53.53434,
      longitude: 9.998654,
      zoom: 16
    },
    id: 11
  },
  {
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
    ],
    title: `Tile House`,
    isFavorite: false,
    rating: 2.9,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 5,
    price: 266,
    goods: [
      `Air conditioning`,
      `Baby seat`,
      `Towels`,
      `Washer`,
      `Breakfast`,
      `Fridge`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    location: {
      latitude: 48.86561,
      longitude: 2.350499,
      zoom: 16
    },
    id: 87
  },
  {
    city: {
      name: `Brussels`,
      location: {
        latitude: 50.846558,
        longitude: 4.351697,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
    ],
    title: `The house among olive `,
    isFavorite: true,
    isPremium: false,
    rating: 4.6,
    type: `apartment`,
    bedrooms: 4,
    maxAdults: 7,
    price: 218,
    goods: [
      `Washer`,
      `Laptop friendly workspace`,
      `Breakfast`,
      `Air conditioning`,
      `Baby seat`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 50.86556,
      longitude: 4.371697,
      zoom: 16
    },
    id: 3
  },
  {
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938362,
        longitude: 6.959974,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`
    ],
    title: `Nice, cozy, warm big bed apartment`,
    isFavorite: true,
    isPremium: false,
    rating: 2.5,
    type: `house`,
    bedrooms: 2,
    maxAdults: 6,
    price: 485,
    goods: [
      `Breakfast`,
      `Washer`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
    location: {
      latitude: 50.945362,
      longitude: 6.935974,
      zoom: 16
    },
    id: 34
  }
];

export const cities = {
  Amsterdam: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  Brussels: {
    name: `Brussels`,
    location: {
      latitude: 50.846558,
      longitude: 4.351697,
      zoom: 13
    }
  },
  Cologne: {
    name: `Cologne`,
    location: {
      latitude: 50.938362,
      longitude: 6.959974,
      zoom: 13
    }
  },
  Dusseldorf: {
    name: `Dusseldorf`,
    location: {
      latitude: 51.225403,
      longitude: 6.776314,
      zoom: 13
    }
  },
  Hamburg: {
    name: `Hamburg`,
    location: {
      latitude: 53.550343,
      longitude: 10.000654,
      zoom: 13
    }
  },
  Paris: {
    name: `Paris`,
    location: {
      latitude: 48.85661,
      longitude: 2.351499,
      zoom: 13,
    }
  }
};


export const reviews = [
  {
    id: 18,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `вапвпапвапвавапва вапвпапвапвавапвавапвпапвапвавапвавапвпапвапвавапвавапвпапвапвавапва`,
    date: `2020-11-24T14:44:02.210Z`
  },
  {
    id: 17,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `Test1 To submit review please make sure to setratingand describe your stay with at least50 characters.\nTo submit review please make sure to setratingand describe your stay with at least50 characters.\n\n\n`,
    date: `2020-11-22T01:32:42.268Z`
  },
  {
    id: 16,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 2,
    comment: `234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123234324324234234221332323213123123123123`,
    date: `2020-11-22T01:32:15.744Z`
  },
  {
    id: 15,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 3,
    comment: `sadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadass`,
    date: `2020-11-22T01:20:16.707Z`
  },
  {
    id: 14,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `sadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsadasadsadsadsada`,
    date: `2020-11-22T01:18:54.165Z`
  },
  {
    id: 13,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `efeffqw efeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqwefeffqw`,
    date: `2020-11-21T16:15:10.563Z`
  },
  {
    id: 12,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 3,
    comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk. `,
    date: `2020-11-21T16:09:20.677Z`
  },
  {
    id: 11,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312123123121231231212312312`,
    date: `2020-11-21T16:08:28.941Z`
  },
  {
    id: 10,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 2,
    comment: `wqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwewqeqweqweqweqwewqeqwe`,
    date: `2020-11-21T16:07:46.202Z`
  },
  {
    id: 9,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `textFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextFieldtextField`,
    date: `2020-11-21T16:07:00.223Z`
  },
  {
    id: 8,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 4,
    comment: `wqewqewqewqewqewqeqweqweqwewqewqeqweqweqweqweqweqweqweqweqweqweqweqweqweqwewqeqweqweqweqwewqeqweqw`,
    date: `2020-11-21T16:05:44.050Z`
  },
  {
    id: 7,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `sdsfsdfdsfsdfdsfsdfdsf`,
    date: `2020-11-21T15:37:35.160Z`
  },
  {
    id: 6,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `qweqweqweqweqweqweqdsdsdsdqwdqwdweqweqweqweqwdwqddwdwq`,
    date: `2020-11-21T14:48:36.841Z`
  },
  {
    id: 5,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `qweqweqweqweqweqweqwewqeqwedweweqewqewqewqewqeqwewqwq`,
    date: `2020-11-21T14:47:26.390Z`
  },
  {
    id: 4,
    user: {
      id: 1,
      isPro: false,
      name: `r-sasha`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`
    },
    rating: 5,
    comment: `fsfsfsd`,
    date: `2020-11-16T17:26:36.123Z`
  },
  {
    id: 3,
    user: {
      id: 15,
      isPro: false,
      name: `Kendall`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`
    },
    rating: 3,
    comment: `I stayed here for one night and it was an unpleasant experience.`,
    date: `2020-11-01T13:38:44.752Z`
  },
  {
    id: 2,
    user: {
      id: 12,
      isPro: true,
      name: `Isaac`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`
    },
    rating: 3,
    comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
    date: `2020-11-01T13:38:44.752Z`
  },
  {
    id: 1,
    user: {
      id: 14,
      isPro: true,
      name: `Corey`,
      avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`
    },
    rating: 5,
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-11-01T13:38:44.752Z`
  }
];


export const user = {
  authorizationStatus: `AUTH`,
  data: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/8.jpg`,
    email: `r-sasha@list.ru`,
    id: 1,
    isPro: false,
    name: ``
  },
  error: ``
};


export const titleInputs = [
  `terribly`,
  `badly`,
  `not bad`,
  `good`,
  `perfect`
];


export const optionList = {
  FILTER_POPULAR: `Popular`,
  FILTER_TOP_RATED_FIRST: `Top rated first`,
  FILTER_PRICE_LOW_TO_HIGH: `Price: low to high`,
  FILTER_PRICE_HIGH_TO_LOW: `Price: high to low`
};


export const baseFilter = `Popular`;
