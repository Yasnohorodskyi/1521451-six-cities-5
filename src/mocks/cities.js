export default [
  {
    id: `1`,
    title: `Amsterdam`,
    link: `Amsterdam`,
    lat: 52.3809453943508,
    lon: 4.931309666406198
  },
  {
    id: `2`,
    title: `Hamburg`,
    link: `Hamburg`,
    lat: 53.5611,
    lon: 9.9933
  },
  {
    id: `3`,
    title: `Dusseldorf`,
    link: `Dusseldorf`,
    lat: 51.2277,
    lon: 6.7735
  },
  {
    id: `4`,
    title: `Brussels`,
    link: `Brussels`,
    lat: 50.8503,
    lon: 4.3517
  },
  {
    id: `5`,
    title: `Cologne`,
    link: `Cologne`,
    lat: 50.9375,
    lon: 6.9603
  },
  {
    id: `6`,
    title: `Paris`,
    link: `Paris`,
    lat: 48.8566,
    lon: 2.3522
  },
];
