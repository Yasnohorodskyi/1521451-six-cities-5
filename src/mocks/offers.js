export default [
  {
    id: 1,
    title: `Beautiful & luxurious studio at great location`,
    gallary: [
      {
        id: 11,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 12,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 13,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 14,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 15,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 16,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: true,
    houseLevel: `Apartment`,
    rating: 4.8,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Apartment`
      },
      {
        field: `Count`,
        value: `3 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 1 adults`
      }
    ],
    prices: `€380`,
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    host: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
      river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
        `An independent House, strategically located between
      Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    }
  },
  {
    id: 2,
    title: `Wood and stone place`,
    gallary: [
      {
        id: 21,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 22,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 23,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 24,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 25,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 26,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 3.8,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `1 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 1 adults`
      }
    ],
    prices: `€80`,
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    host: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    }
  },
  {
    id: 3,
    title: `Nice, cozy, warm big bed apartment`,
    gallary: [
      {
        id: 31,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 32,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 33,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 34,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 35,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 36,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: `€180`,
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    host: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    }
  },
  {
    id: 4,
    title: `Canal View Prinsengracht`,
    gallary: [
      {
        id: 41,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 42,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 43,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 44,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 45,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 46,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: `€180`,
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    host: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    },
  },
  {
    id: 5,
    title: `Nice, cozy, warm big bed apartment`,
    gallary: [
      {
        id: 51,
        src: `/img/apartment-02.jpg`,
        alt: `Photo studio`
      },
      {
        id: 52,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 53,
        src: `/img/room.jpg`,
        alt: `Photo studio`
      },
      {
        id: 54,
        src: `/img/apartment-03.jpg`,
        alt: `Photo studio`
      },
      {
        id: 55,
        src: `/img/studio-01.jpg`,
        alt: `Photo studio`
      },
      {
        id: 56,
        src: `/img/apartment-01.jpg`,
        alt: `Photo studio`
      },
    ],
    premium: false,
    houseLevel: `Private room`,
    rating: 4.9,
    propertyFeatures: [
      {
        field: `Type`,
        value: `Room`
      },
      {
        field: `Count`,
        value: `5 Bedrooms`
      },
      {
        field: `MaxPeople`,
        value: `Max 4 adults`
      }
    ],
    prices: `€180`,
    insides: {
      1: `Wi-Fi`,
      2: `Heating`,
      3: `Kitchen`,
      4: `Fridge`,
      5: `Washing machine`,
      6: `Coffee machine`,
      7: `Dishwasher`,
      8: `Towels`,
      9: `Baby seat`,
      10: `Cabel TV`
    },
    host: {
      ava: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      description: [
        `A quiet cozy and picturesque that hides behind a a
  river by the unique lightness of Amsterdam.
  The building is green and from 18th century.`,
        `An independent House, strategically located between
  Rembrand Square and National Opera, but where the
  bustle of the city comes to rest in this alley flowery and colorful.`
      ]
    }
  }
];
