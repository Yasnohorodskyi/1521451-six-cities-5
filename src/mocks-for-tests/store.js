import configureStore from "redux-mock-store";
import {offers, cities, reviews, user, baseFilter} from "./mocks";
import {offersAdapter, offerAdapter} from "../helpers/offers-adapter";

const createStore = configureStore();

const mockStore = createStore({
  Offers: {
    currentCity: cities[`Brussels`],
    baseFilter,
    offers: offersAdapter(offers),
    offer: offerAdapter(offers[0]),
    nearby: offersAdapter(offers),
    favorites: offersAdapter(offers),
    data: offersAdapter(offers),
    getFavoriteDispatch: ()=>{}
  },
  Review: {
    reviews
  },
  User: user,
  Offer: offerAdapter(offers[0])
});

export {mockStore};
