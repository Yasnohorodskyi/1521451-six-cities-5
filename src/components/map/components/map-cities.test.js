import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {MapCities} from "./map-cities";
import {offers, cities} from "../../../mocks-for-tests/mocks";

it(`MapCities is rendered correctly`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <MapCities
        offers={offers}
        currentCity={cities[`Brussels`]}
      />
  );
  expect(tree).toMatchSnapshot();
});
