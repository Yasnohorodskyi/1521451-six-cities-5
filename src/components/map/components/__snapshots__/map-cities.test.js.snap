// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapCities is rendered correctly 1`] = `
<div
  id="map"
>
  <Map
    animate={true}
    doubleClickZoom={true}
    dragging={true}
    easeLinearity={0.35}
    scrollWheelZoom={true}
  >
    <TileLayer
      attribution="Â© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
      url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"
    />
  </Map>
</div>
`;
